### DISCUSSION TITLE
Will not pop up on network

### DISCUSSION THREAD

**Discussion by Finagain** [emporia-vue-local/esphome#26] [Q&A] [answered]
I did everything suggested (Static IP, connected to mains [Discussion 8](https://github.com/emporia-vue-local/esphome/discussions/8)) and it still isn't showing up neither on the router or in ESPHome. I thought it might be a bad unit but for fun, I re-flashed with the original firmware and it popped up on the network. I can't get ESPHome to work on this thing. I even tried updating my ESPhome to 2022.8.2 and then down to 2022.1.2 (the same version the guy uses on YouTube) and nothing works. I have used ESPHome web flasher, NodeMCU Py flasher and both flashed successful (reflashed with original). I'm not new to ESPhome and have several devices working currently. What version of ESPhome are you guys working with?

**Comment by flaviut** [comment]
I'm using 2022.5.1.

I don't like guessing about what's going on; the logs should explain what's going on.

Can you view the logs when you connect your Vue directly to your computer? Not all serial adapters are capable of providing enough power to the Vue.

If not, you can try using a dedicated 5V power supply and then look at the logs. You'll need to tie your serial adapter's ground to the 5V power supply's ground.

If you can't/won't do that, and you promise not to sue me (this stuff can easily kill you!), you can take a more dangerous approach:

- unplug the vue from mains
- connect the serial adapter, leaving the 5V & 3.3V connections unconnected
- tape the vue down to your desk or otherwise make sure it won't move
- **DO NOT TOUCH THE VUE PAST THIS POINT**
- plug in the vue to mains
- save the serial log output
- unplug the vue from the mains
- you can touch it now

(note: this only works because the low voltage side of the vue is isolated from mains. do not ever try this with any other device, you'll fry your computer)

**Reply by Finagain** [reply]
Thank you for all of the suggestions.  I haven't been able to get any logs from it yet.  I will try your version of ESPHome first (easiest) and then when my lawyer gets back to me :) I'll try the other suggestion!  Thanks so much for the help.  I'll update when I get something.

**Reply by Finagain** [reply]
Connected to mains. RXD TO RXD TXD TO TXD on USB adapter.  Nothing else connected.
This is all it keeps looping:

`rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371 
ets Jun  8 2016 00:22:57`

**Reply by flaviut** [reply]
Could be brown-out if your serial adapter can't supply enough power. I'd suggest powering it separately to see if you get a different message.

**Reply by Finagain** [reply]
I edited my message at the same time.  The Emporia is connected to main power.

**Reply by flaviut** [reply]
Hmm. Seems like the flash is corrupted--there's a lot of people on the internet complaining about this issue, but it seems to usually be them designing the hardware wrong.

Make sure you're not connected to mains at this point

Are you using esptool to flash the data? Maybe try manually clearing the flash first, and then write it:

```
$ esptool.py --port ... erase_flash
$ esptool.py --port ... write_flash 0x0 vue2.bin
```

Do you get any interesting messages from those commands?

Maybe also inspect the board carefully for any bridged pins--although that doesn't explain why the default firmware works fine :thinking:

**Reply by Finagain** [reply]
My main mode of flashing is NodeMCU PyFlasher by I will try esptool.py.  Thanks again for your time!

**Reply by flaviut** [reply]
Looks like there's an erase flash button there--try checking it before doing an upload. Pyflasher is based on esptool.py and does exactly the same thing.

**Comment by Finagain** [comment]
I always have that set to Yes.  

Here's the last bit from the compile:

`Linking .pioenvs/emporiavue2/bootloader.elf
Building .pioenvs/emporiavue2/bootloader.bin
Linking .pioenvs/emporiavue2/firmware.elf
RAM:   [=         ]  10.5% (used 34256 bytes from 327680 bytes)
Flash: [====      ]  40.7% (used 746505 bytes from 1835008 bytes)
Building .pioenvs/emporiavue2/firmware.bin
esp32_create_combined_bin([".pioenvs/emporiavue2/firmware.bin"], [".pioenvs/emporiavue2/firmware.elf"])
Wrote 0xc6590 bytes to file /config/.esphome/build/emporiavue2/.pioenvs/emporiavue2/firmware-factory.bin, ready to flash to offset 0x0
======================== [SUCCESS] Took 390.70 seconds ========================
INFO Successfully compiled program.`

Here is the flasher:
![Image 9-5-22 at 10 56 AM](https://user-images.githubusercontent.com/71937166/188501042-d2bcdf91-f538-4fe2-8fed-873aec25ee30.jpg)

**Reply by flaviut** [reply]
Honestly, I have no idea. This all looks fine to me.

The only other things I can suggest are

- flashing without clearing the flash first
- if you haven't already, try clearing cached files and then compiling and flashing
- maybe use a different serial dongle (doubt it, I would have expected an error in the flashing process)
- maybe it's defective? Maybe you can return the original firmware into it and return it?

**Reply by Finagain** [reply]
I just reflashed the factory firmware again and it booted right up.  Super weird. I'm going to try a few more things before I give up.

**Comment by Finagain** [comment]
And I did a fresh compile as I spun up a 2022.5.1 version of ESPHome.

**Comment by Finagain** [comment]
Not sure if this is a factor but my chip is ESP32-WROVER-1B and yours is ESP32-WROVER-B.

**Comment by flaviut** [comment]
The only difference between the two models is that the -B has no external antenna port and the -IB does, so in my case, it was probably just a misprinted part.

**Reply by Finagain** [reply]
Sorry to waste your time.  I could have just googled that...

**Comment by PanicRide** [comment]
I too had an Emporia device that flashed correctly but refused to connect for some reason.

I had to carefully go through the precarious situation of taking it apart while connected to mains and connecting the serial adapter so I could monitor the log.

Even though my wifi network showed up in its scan list, it was saying the configured network wasn't found. I have no idea why. 🤷

Since the device is in a fixed position it makes sense to turn on wifi fast connect so that it constantly attempts to connect to the network without bothering to scan for it first. This ended up working and it connects just fine now.

![Screenshot_20220906-112743](https://user-images.githubusercontent.com/99451581/188713208-cf7453b9-3b99-4a84-bc9f-58eb7886c575.png)

**Reply by Finagain** [reply]
I just tried the fast connect but am still getting this error while monitoring the serial connection.

`rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371 `

Is it possible that it is referring to line 371 of file ets_main.c?

**Comment by Finagain** [comment]
Nevermind. I'm  dumb...  I did find this though.

![Image 9-6-22 at 12 38 PM](https://user-images.githubusercontent.com/71937166/188723848-80319644-0af8-4a9d-80f2-2b4cde50d723.jpg)

**Comment by Finagain** [comment, marked as answer]
I GOT IT!!!  re-googled my first error code and found someone tried to write at 0x1000 so I did that.  Didn't work but I got a different error and that led me to someone that went from NodeMCU Pyflasher to ESPHome-Flasher.  THAT'S WHAT WORKED... 

THANK YOU for all of your time and help!

**Reply by flaviut** [reply]
What the hell :grinning: 

Glad you figured it out! I have no idea why ESPHome-flasher would work when pyflasher would not--that's very strange.

Thank you for letting me know that you got it working!

**Reply by Finagain** [reply]
Bought a second one and tried to use ESPHome-Flasher directly but got an SLA Certificate warning.  Flashed with NodeMCU Pyflasher first (still didn't connect) and THEN ESPHome-Flasher and it worked like a charm...  Don't know why but took 20 minutes start to finish vs 2 days.
